src/chunk.js

```diff
@@ -393,17 +408,21 @@ class Chunks {
    * @private
    */
   execute_(idleDeadline) {
+    this.scheduledImmediateInvocation_ = false;
     const t = this.nextTask_(/* opt_dequeue */ true);
     if (!t) {
       return false;
     }
-    const before = Date.now();
-    this.timeSinceLastExecution_ = before;
-    t.runTask_(idleDeadline);
-    resolved.then(() => {
-      this.schedule_();
-    });
-    dev().fine(TAG, t.getName_(), 'Chunk duration', Date.now() - before);
+    try {
+      const before = Date.now();
```

Comment:
What is the motivation for moving this logic into a try/finally?

Reply:
With the new logic if we fail to schedule a next run after completing a run we risk never to unwind the rest of the queue. This makes it locally understandable that we will always schedule again even if an exception is thrown.